salt/core/sshd.sls

```diff
@@ -0,0 +1,29 @@
+# We'll only be using SSH key authentication
+disable password authentication:
+  file.replace:
+    - name: /etc/ssh/sshd_config
+    - pattern: "^#?PasswordAuthentication .*"
+    - repl: PasswordAuthentication no
+
+# Technically not needed but setting for posterity
+force root ssh keys:
+  file.replace:
+    - name: /etc/ssh/sshd_config
+    - pattern: "^#?PermitRootLogin.*"
+    - repl: "PermitRootLogin without-password"
```

Comment:
If not needed, can we delete it?

Reply:
We'd like to keep this setting so that it is clear to users that you cannot log in to root without a password.
As I said in my comment we're already technically doing this by also setting with `PasswordAuthentication no`. 

There are two reasons for doing this, firstly it adds an extra layer to the security if `PasswordAuthentication` is toggled back on, the second reason is that while `PermitRootLogin` is set to either `no` or `without-password` we can monitor it with our intrusion detection software.

I think we should update the comment to show this, how do you think it would be best to phrase it?

Reply:
I think your 2nd-3rd sentences explain it well in a comment :) Feel free to copy-edit as desired, update and merge.

Reply:
Comment improved.